.github/workflows/ci.yml

```diff
@@ -80,134 +80,152 @@ jobs:
     name: Build docs
     runs-on: ubuntu-latest
     env:
-      TRAVIS_JOB_NAME: "Build documentation"
       PYTHON_VERSION: 3.6
+      CI_JOB_TYPE: "Documentation"
     steps:
       - uses: actions/checkout@master
-      - name: "Build documentation"
+      - name: "Build CI image"
+        run: ./scripts/ci/ci_prepare_image_on_ci.sh
+      - name: "Build docs"
         run: ./scripts/ci/ci_docs.sh
 
   tests-p36-postgres-integrations:
-    name: "Tests [Postgres9.6][Py3.6][integrations]"
+    name: "[Pg9.6][Py3.6][integrations]"
     runs-on: ubuntu-latest
     needs: [statics, statics-tests]
     env:
-      TRAVIS_JOB_NAME: "Tests [Postgres9.6][Py3.6][integrations]"
       BACKEND: postgres
       PYTHON_VERSION: 3.6
       POSTGRES_VERSION: 9.6
       ENABLED_INTEGRATIONS: "cassandra kerberos mongo openldap rabbitmq redis"
       RUN_INTEGRATION_TESTS: all
+      CI_JOB_TYPE: "Tests"
     steps:
       - uses: actions/checkout@master
-      - name: "Tests [Postgres9.6][Py3.6][integrations]"
+      - name: "Build CI image"
+        run: ./scripts/ci/ci_prepare_image_on_ci.sh
+      - name: "Tests"
         run: ./scripts/ci/ci_run_airflow_testing.sh
 
   tests-p36-postgres-providers:
```

Comment:
We can do all of this using a matrix

```
strategy:
  matrix:
    postgres: [Pg9, Pg10]
    python: [3.6, 3.7]
    strategy: [providers, core]
```

Reply:
Actually I guess we have far less limits now for GitHub actions so rather than splitting the tests we can combine the tests back.

Reply:
This will limit number of concurrent jobs, so quite a good idea

Reply:
Working on it. I will also (I think) incorporate a great idea from @turbaszek -> It turns out that (similarly as in GitLab POC I tried before) each project has a private GitHub Docker registry that we can use to store the images, so for every build we will just build one CI image with latest sources and use it for all the jobs. I discussed it with Tomek this morning and I thought a bit about it and I can do it very easily now within this CI "optimisation" build.

Bear with me. I will finish it today :).